File: GNSDK/wrappers/gnsdk_java/src_wrapper/j2se/com/gracenote/gnsdk/GnResponseTracks.java

package com.gracenote.gnsdk;

/** 
* Collection of track results received in response to a track query 
*/ 
 
public class GnResponseTracks extends GnDataObject {
  private long swigCPtr;

  protected GnResponseTracks(long cPtr, boolean cMemoryOwn) {
    super(gnsdk_javaJNI.GnResponseTracks_SWIGUpcast(cPtr), cMemoryOwn);
    swigCPtr = cPtr;
  }

  protected static long getCPtr(GnResponseTracks obj) {
    return (obj == null) ? 0 : obj.swigCPtr;
  }

  protected void finalize() {
    delete();
  }

  public synchronized void delete() {
    if (swigCPtr != 0) {
      if (swigCMemOwn) {
        swigCMemOwn = false;
        gnsdk_javaJNI.delete_GnResponseTracks(swigCPtr);
      }
      swigCPtr = 0;
    }
    super.delete();
  }

  public static String gnType() {
    return gnsdk_javaJNI.GnResponseTracks_gnType();
  }

  public static GnResponseTracks from(GnDataObject obj) throws com.gracenote.gnsdk.GnException {
    return new GnResponseTracks(gnsdk_javaJNI.GnResponseTracks_from(GnDataObject.getCPtr(obj), obj), true);
  }

/** 
*  Result count - number of matches returned 
*  @return Count 
*/ 
 
  public long resultCount() {
    return gnsdk_javaJNI.GnResponseTracks_resultCount(swigCPtr, this);
  }

/** 
*  Range start - ordinal value of first match in range total 
*  @return Range start ordinal 
*/ 
 
  public long rangeStart() {
    return gnsdk_javaJNI.GnResponseTracks_rangeStart(swigCPtr, this);
  }

/** 
*  Range end - ordinal value of last match in range total 
*  @return Range end ordinal 
*/ 
 
  public long rangeEnd() {
    return gnsdk_javaJNI.GnResponseTracks_rangeEnd(swigCPtr, this);
  }

/** 
*  Range total - total number of matches that could be returned 
*  @return Range total 
*/ 
 
  public long rangeTotal() {
    return gnsdk_javaJNI.GnResponseTracks_rangeTotal(swigCPtr, this);
  }

/** 
*  Flag indicating if response needs user decision - either multiple matches returned or less than perfect single match 
*  @return True if user decision required, false otherwise 
*/ 
 
  public boolean needsDecision() {
    return gnsdk_javaJNI.GnResponseTracks_needsDecision(swigCPtr, this);
  }

  public GnTrackIterable tracks() {
    return new GnTrackIterable(gnsdk_javaJNI.GnResponseTracks_tracks(swigCPtr, this), true);
  }

}

File: GNSDK/wrappers/gnsdk_csharp/src_wrapper/GnResponseTracks.cs

namespace GracenoteSDK {

using System;
using System.Runtime.InteropServices;

/**
* Collection of track results received in response to a track query
*/
public class GnResponseTracks : GnDataObject {
  private HandleRef swigCPtr;

  internal GnResponseTracks(IntPtr cPtr, bool cMemoryOwn) : base(gnsdk_csharp_marshalPINVOKE.GnResponseTracks_SWIGUpcast(cPtr), cMemoryOwn) {
    swigCPtr = new HandleRef(this, cPtr);
  }

  internal static HandleRef getCPtr(GnResponseTracks obj) {
    return (obj == null) ? new HandleRef(null, IntPtr.Zero) : obj.swigCPtr;
  }

  ~GnResponseTracks() {
    Dispose();
  }

  public override void Dispose() {
    lock(this) {
      if (swigCPtr.Handle != IntPtr.Zero) {
        if (swigCMemOwn) {
          swigCMemOwn = false;
          gnsdk_csharp_marshalPINVOKE.delete_GnResponseTracks(swigCPtr);
        }
        swigCPtr = new HandleRef(null, IntPtr.Zero);
      }
      GC.SuppressFinalize(this);
      base.Dispose();
    }
  }

  public static string GnType() {
    string ret = gnsdk_csharp_marshalPINVOKE.GnResponseTracks_GnType();
    return ret;
  }

  public static GnResponseTracks From(GnDataObject obj) {
    GnResponseTracks ret = new GnResponseTracks(gnsdk_csharp_marshalPINVOKE.GnResponseTracks_From(GnDataObject.getCPtr(obj)), true);
    if (gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Pending) throw gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Retrieve();
    return ret;
  }

/**
*  Result count - number of matches returned
*  @return Count
*/
  public uint ResultCount() {
    uint ret = gnsdk_csharp_marshalPINVOKE.GnResponseTracks_ResultCount(swigCPtr);
    return ret;
  }

/**
*  Range start - ordinal value of first match in range total
*  @return Range start ordinal
*/
  public uint RangeStart() {
    uint ret = gnsdk_csharp_marshalPINVOKE.GnResponseTracks_RangeStart(swigCPtr);
    return ret;
  }

/**
*  Range end - ordinal value of last match in range total
*  @return Range end ordinal
*/
  public uint RangeEnd() {
    uint ret = gnsdk_csharp_marshalPINVOKE.GnResponseTracks_RangeEnd(swigCPtr);
    return ret;
  }

/**
*  Range total - total number of matches that could be returned
*  @return Range total
*/
  public uint RangeTotal() {
    uint ret = gnsdk_csharp_marshalPINVOKE.GnResponseTracks_RangeTotal(swigCPtr);
    return ret;
  }

/**
*  Flag indicating if response needs user decision - either multiple matches returned or less than perfect single match
*  @return True if user decision required, false otherwise
*/
  public bool NeedsDecision() {
    bool ret = gnsdk_csharp_marshalPINVOKE.GnResponseTracks_NeedsDecision(swigCPtr);
    return ret;
  }

  public GnTrackEnumerable Tracks {
    get {
      IntPtr cPtr = gnsdk_csharp_marshalPINVOKE.GnResponseTracks_Tracks_get(swigCPtr);
      GnTrackEnumerable ret = (cPtr == IntPtr.Zero) ? null : new GnTrackEnumerable(cPtr, true);
      return ret;
    } 
  }

}

}
